test/new-e2e/amd.spec.js

```diff
@@ -0,0 +1,29 @@
+/* global Treasure */
+var expect = require('expect.js')
+
+describe('amd', function () {
+  it('should be compatible with require.js', function () {
+    browser.url('http://localhost:1337/fixtures/amd/index.html')
+
+    // RequireJS Module should load an execute correctly
+    browser.waitUntil(function () {
+      return browser.getText('#rjs') === 'success'
+    }, 5000, 'expected Require.js status to be success after 5s')
+
+    browser.timeouts('script', 5000)
+    var result = browser.executeAsync(function (done) {
```

Comment:
Does this need to be awaited somehow?  The Nightwatch documentation seems to be unclear AFAICT about the API around this -- there's apparently a callback you can pass in to it to regain control.

Reply:
It's wdio http://webdriver.io/api/protocol/executeAsync.html

And elsewhere in the docs "Different than using the standalone mode all commands that get executed by the wdio test runner are synchronous."

Reply:
But how can that possibly be synchronous if it returns a native Promise?
https://github.com/webdriverio/webdriverio/blob/master/lib/utils/RequestHandler.js#L204
https://github.com/webdriverio/webdriverio/blob/master/lib/utils/RequestHandler.js#L162
and even the source handles it with a `.catch`?
https://github.com/webdriverio/webdriverio/blob/master/lib/protocol/executeAsync.js#L67
😕

Reply:
```
// Per default WebdriverIO commands getting executed in a synchronous way using
// the wdio-sync package. If you still want to run your tests in an async way
// using promises you can set the sync command to false.
```
http://webdriver.io/guide/testrunner/configurationfile.html

We can disable it by setting `"async": true` in the config.

Reply:
But that deals directly with native Promises, too?
https://github.com/webdriverio/wdio-sync/blob/master/index.js#L513